requirements.txt

```diff
@@ -3,3 +3,4 @@ matplotlib
 pillow
 sphinx
 nose
+nbformat
```

Comment:
nbformat pulls in traitlets, ipython_genutils, jsonchema, and jupyter_core.

I worry particularly about jupyter_core: we could easily run into conflicts or break an IPython install by dragging in an incompatible version.

It seems like a big cost, given that writing a notebook is not much more than writing a json file. How difficult would it be to have our own code for writing such a file?